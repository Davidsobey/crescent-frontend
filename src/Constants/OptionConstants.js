const OptionConstants = {
  CREATE_REQUEST: 'CREATE_OPTION_REQUEST',
  CREATE_SUCCESS: 'CREATE_OPTION_SUCCESS',
  CREATE_FAILURE: 'CREATE_OPTION_FAILURE',

  GETALL_REQUEST: 'GETALL_OPTION_REQUEST',
  GETALL_SUCCESS: 'GETALL_OPTION_SUCCESS',
  GETALL_FAILURE: 'GETALL_OPTION_FAILURE',

  GETBYID_REQUEST: 'GETBYID_OPTION_REQUEST',
  GETBYID_SUCCESS: 'GETBYID_OPTION_SUCCESS',
  GETBYID_FAILURE: 'GETBYID_OPTION_FAILURE',
};

export default OptionConstants;
