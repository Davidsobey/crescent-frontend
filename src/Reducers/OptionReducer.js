import OptionConstants from '../Constants/OptionConstants';
import UserConstants from '../Constants/UserConstants';

function OptionReducer(state = {}, action) {
  switch (action.type) {
    case OptionConstants.CREATE_REQUEST:
      return {
        Options: action.OptionName,
      };
    case OptionConstants.CREATE_SUCCESS:
      return {
        Options: action.OptionName,
      };
    case OptionConstants.CREATE_FAILURE:
      return {};
    case OptionConstants.GETALL_REQUEST:
      return {
        loading: true,
      };
    case OptionConstants.GETALL_SUCCESS:
      return Object.assign({}, state, {
        Options: action.Options,
      });
    case OptionConstants.GETALL_FAILURE:
      return {
        Options: {},
      };
    case OptionConstants.GETBYID_REQUEST:
      return {
        loading: true,
      };
    case OptionConstants.GETBYID_SUCCESS:
      return Object.assign({}, state, {
        Options: action.Options,
      });
    case OptionConstants.GETBYID_FAILURE:
      return {
        Options: {},
      };
    case UserConstants.LOGOUT:
      return {};
    default:
      return state;
  }
}

export default OptionReducer;
