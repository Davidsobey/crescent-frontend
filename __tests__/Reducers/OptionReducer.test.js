// Reducer to be tested
import OptionReducer from '../../src/Reducers/OptionReducer';
import OptionConstants from '../../src/Constants/OptionConstants';

describe('OptionReducer', () => {
  test('GETALL_SUCCESS', () => {
    const action = { 
      type: OptionConstants.GETALL_SUCCESS, 
      Options: [],
    };

    expect(OptionReducer(undefined, action)).toMatchSnapshot();
  });

  test('GETBYID_SUCCESS', () => {
    const action = { 
      type: OptionConstants.GETBYID_SUCCESS, 
      Options: [],
    };

    expect(OptionReducer(undefined, action)).toMatchSnapshot();
  });
});
