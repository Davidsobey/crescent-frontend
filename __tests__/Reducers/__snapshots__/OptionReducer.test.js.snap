// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OptionReducer GETALL_SUCCESS 1`] = `
Object {
  "Options": Array [],
}
`;

exports[`OptionReducer GETBYID_SUCCESS 1`] = `
Object {
  "Options": Array [],
}
`;
